src/solvers/dgmulti/flux_differencing_gauss_sbp.jl

```diff
@@ -25,15 +25,262 @@ function tensor_product_quadrature(element_type::Hex, r1D, w1D)
+    end
+  end
+end
+
+# Interpolates values from volume Gauss nodes to face nodes on one element.
+@inline function tensor_product_gauss_face_operator!(out::AbstractVector,
+                                                     A::TensorProductGaussFaceOperator{3, Interpolation},
+                                                     x::AbstractVector)
+
+  @unpack interp_matrix_gauss_to_face_1d, face_indices_tensor_product = A
+  @unpack nnodes_1d, nfaces = A
+
+  fill!(out, zero(eltype(out)))
+
+  # for 3D GaussSBP nodes, the indexing is first in y, then x, then z.
+  x = reshape(x, nnodes_1d, nnodes_1d, nnodes_1d)
+
+  # interpolation in the y-direction
+  @turbo for i in Base.OneTo(nnodes_1d), j in Base.OneTo(nnodes_1d) # loop over nodes in a face
+    index_left = face_indices_tensor_product[1, i, j, 2]
+    index_right = face_indices_tensor_product[2, i, j, 2]
+    for jj in Base.OneTo(nnodes_1d) # loop over "line" of volume nodes
+      out[index_left] = out[index_left] + interp_matrix_gauss_to_face_1d[1, jj] * x[jj, i, j]
+      out[index_right] = out[index_right] + interp_matrix_gauss_to_face_1d[2, jj] * x[jj, i, j]
+    end
+  end
+
+  # interpolation in the x-direction
+  @turbo for i in Base.OneTo(nnodes_1d), j in Base.OneTo(nnodes_1d) # loop over nodes in a face
+    index_left = face_indices_tensor_product[1, i, j, 1]
+    index_right = face_indices_tensor_product[2, i, j, 1]
+    for jj in Base.OneTo(nnodes_1d) # loop over "line" of volume nodes
+      out[index_left] = out[index_left] + interp_matrix_gauss_to_face_1d[1, jj] * x[i, jj, j]
+      out[index_right] = out[index_right] + interp_matrix_gauss_to_face_1d[2, jj] * x[i, jj, j]
+    end
+  end
+
+  # interpolation in the z-direction
+  @turbo for i in Base.OneTo(nnodes_1d), j in Base.OneTo(nnodes_1d) # loop over nodes in a face
+    index_left = face_indices_tensor_product[1, i, j, 3]
+    index_right = face_indices_tensor_product[2, i, j, 3]
+    for jj in Base.OneTo(nnodes_1d) # loop over "line" of volume nodes
+      # the ordering (i,j) -> (j,i) needs to be reversed for this last face.
+      # this is probably due to way we define face nodes for Hex() types in StartUpDG.jl.
+      out[index_left] = out[index_left] + interp_matrix_gauss_to_face_1d[1, jj] * x[j, i, jj]
+      out[index_right] = out[index_right] + interp_matrix_gauss_to_face_1d[2, jj] * x[j, i, jj]
+    end
+  end
+end
+
+# Projects face node values to volume Gauss nodes on one element.
+@inline function tensor_product_gauss_face_operator!(out_vec::AbstractVector,
+                                                     A::TensorProductGaussFaceOperator{2, Projection},
+                                                     x::AbstractVector)
+
+  @unpack interp_matrix_gauss_to_face_1d, face_indices_tensor_product = A
+  @unpack inv_volume_weights_1d, nnodes_1d, nfaces = A
+
+  fill!(out_vec, zero(eltype(out_vec)))
+  out = reshape(out_vec, nnodes_1d, nnodes_1d) # @turbo appears to break for reshaped arrays
```

Comment:
```suggestion

  # for 2D GaussSBP nodes, the indexing is first in x, then y
  out = reshape(out_vec, nnodes_1d, nnodes_1d)
```
Is the comment on `@turbo` and `reshape` still true? I'm asking since you use `@turbo` below...

Reply:
No, it doesn't appear to be an issue anymore - the comment is incorrect.